test/Main.hs

```diff
@@ -227,24 +227,76 @@ assertEqual preface expected actual eq =
 
 ----------------------------------------
 
+sqlGenInts :: Int32 -> SQL
+sqlGenInts n = smconcat
+  [ "WITH RECURSIVE ints(n) AS"
+  , "( VALUES (1) UNION ALL SELECT n+1 FROM ints WHERE n <" <?> n
+  , ") SELECT n FROM ints"
+  ]
+
+cursorTest :: TestData -> Test
+cursorTest td = testGroup "Cursors"
+  [ basicCursorWorks
+  , scrollableCursorWorks
+  , withHoldCursorWorks
+  ]
+  where
+    basicCursorWorks = testCase "Basic cursor works" $ do
+      runTestEnv td def . withCursor "ints" mempty (sqlGenInts 5) $ \cursor -> do
+        xs <- (`fix` []) $ \loop acc -> cursorFetch cursor CD_Next >>= \case
```

Comment:
I'd just write `let go acc = ... in go []` but it doesn't matter that much.

Reply:
I prefer to use `fix` for recursion with 0/1 arguments, avoids awkward indentation you get with `let` or non-locality of `where`.